tests/device_compilation_tests.cpp

```diff
@@ -171,6 +171,35 @@ struct KernelFunctor {
   AccessorT acc;
 };
 
+BOOST_AUTO_TEST_CASE(submit_zero_work_items) {
+  cl::sycl::queue q;
+  const std::size_t N = 1024;
+
+  auto d_y = static_cast<int *>(cl::sycl::malloc_device(sizeof(int) * N, q));
+
+  q.parallel_for<>(
+    cl::sycl::range<1>(static_cast<std::size_t>(0)),
+    [=](cl::sycl::id<1> idx) {
+      d_y[idx] = static_cast<int>(idx);
```

Comment:
Why are we writing something in this test? Was the intention to also add a test to check that the initial data array was not modified by the 0-size kernel? (in that case, data initialization and result comparison would be missing)